### ISSUE TITLE
Some issues

### ISSUE THREAD

**Issue by creynders** [appersonlabs/Mooring#1]
Hi @mattapperson just browsing through mooring, there's a few small issues though:
- if the middleware passes nothing through or an empty object, the data is not immutable

``` js
hooks.before('cook', false, (food, next)=>{
    food.name='A';
    next();
});

var cook = hooks.createHook('cook', food => {
    console.log(food);//outputs: {name:'A'}
});
```
- you can't pass functions as an argument to the function created with `createHook`

``` js
var cook = hooks.createHook('cook', function(food) {
    console.log(food); //outputs: undefined
});
cook(()=>{});
```

Really curious to see what direction this is going in!

**Comment by creynders** [comment]
I feel a bit awkward asking this, but since grappling-hook is a keystone product not mine, I think I have to: could you follow the MIT license rules of attribution, since there are parts literally taken from grappling-hook?

**Comment by mattapperson** [comment]
@crynders sure thing! I was on the fence as the methods were relatively heavenly modded. So I ended up just calling out the source and giving credit in the readme. Any particular parts you would like credit for?

**Comment by mattapperson** [comment]
Also great catches on those 2 issues, I'll take a look at them today! 😊

**Comment by mattapperson** [comment]
@creynders Credit added and copyright edited. Sorry this was just something I whipped up so I even forgot the copyright... my bad.

**Comment by mattapperson** [comment]
@creynders fixed the data mutability bug, working on the 1 argument being a method bug now

**Comment by mattapperson** [comment]
@creynders using the exact code you provided for "you can't pass functions as an argument to the function created with" I could not reproduce the issue.

**Comment by mattapperson** [comment]
The issue (per a side convo with @creynders) is:

~~~js
hooks.before('cook', false, function(food, next) {
    console.log('cook pre', food); //outputs undefined
    next();
});

var cook = hooks.createHook('cook', function(food) {
    console.log('createHook callback', food);
    return food;
});

cook(function(){});
~~~

**Comment by mattapperson** [comment]
All issues fixed in 1.0.4
